ring/replication_set.go

```diff
@@ -279,6 +292,16 @@ func DoUntilQuorumWithoutSuccessfulContextCancellation[T any](ctx context.Contex
 		}
 	}
 
+	terminate := func(err error) ([]T, error) {
+		if cfg.Logger != nil {
+			_ = cfg.Logger.Error(err)
+		}
```

Comment:
[nit] I think you can just use `logger`.

Reply:
`logger` only supports logging, whereas we want to set the error flag on the span here.